### PR TITLE
Change default metrics address to localhost:8888

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

**Review by songy23 — APPROVED** [review]
👍  this will make macos users' life easier

File: service/telemetry/factory.go [hunk 1]
```diff
@@ -18,9 +18,16 @@ import (
...
+	"telemetry.UseLocalHostAsDefaultMetricsAddress",
+	featuregate.StageBeta,
+	featuregate.WithRegisterFromVersion("v0.111.0"),
+	featuregate.WithRegisterDescription("controls whether default Prometheus metrics server use localhost as the default host for their endpoints"),
+)
+
 // disableHighCardinalityMetricsfeatureGate is the feature gate that controls whether the collector should enable
 // potentially high cardinality metrics. The gate will be removed when the collector allows for view configuration.
-var disableHighCardinalityMetricsfeatureGate = featuregate.GlobalRegistry().MustRegister(
+var disableHighCardinalityMetricsFeatureGate = featuregate.GlobalRegistry().MustRegister(
```
Comment (atoulme) [inline service/telemetry/factory.go:30]: that change is not related

Comment (bogdandrutu) [reply service/telemetry/factory.go:30]: Since I see people here not being responsible I try to put everything in less PRs :))

Comment (bogdandrutu) [reply service/telemetry/factory.go:30]: https://github.com/open-telemetry/opentelemetry-collector/pull/11278

Comment (bogdandrutu) [reply service/telemetry/factory.go:30]: Also, to clarify my joke, I was trying to find an excuse why I did not do the right thing.

**Review by dmitryax — APPROVED** [review]
